api/v01/entities/vrp_input.rb

```diff
@@ -0,0 +1,522 @@
+      use :vrp_request_activity
+    end
+    mutually_exclusive :activity, :activities
+    optional(:quantity_ids, type: String, documentation: { hidden: true }, desc: 'Quantities to consider, CSV front only')
+    optional(:quantities, type: Array, desc: 'Define the entities which are taken or dropped') do
+      use :vrp_request_quantity
+    end
+    mutually_exclusive :quantity_ids, :quantities
+  end
+
+  params :vrp_request_shipment do
+    requires(:id, type: String, allow_blank: false, desc: '')
+    optional(:priority, type: Integer, values: 0..8, desc: 'Priority assigned to the service in case of conflict to assign every jobs (from 0 to 8, default is 4)')
+    optional(:exclusion_cost, type: Integer, desc: 'Exclusion cost')
+
+    optional(:visits_number, type: Integer, coerce_with: ->(val) { val.to_i.positive? && val.to_i }, default: 1, allow_blank: false, desc: 'Total number of visits over the complete schedule (including the unavailable visit indices)')
+
+    optional(:unavailable_visit_indices, type: Array[Integer], desc: '(Scheduling only) unavailable indices of visit')
+
+    optional(:unavailable_visit_day_indices, type: Array[Integer], desc: '(Scheduling only) Express the exceptionnals days indices of unavailabilty')
+    optional(:unavailable_visit_day_date, type: Array, desc: '(Scheduling only) Express the exceptionnals days of unavailability')
+    mutually_exclusive :unavailable_visit_day_indices, :unavailable_visit_day_date
+
+    optional(:minimum_lapse, type: Float, desc: 'Minimum day lapse between two visits')
+    optional(:maximum_lapse, type: Float, desc: 'Maximum day lapse between two visits')
+
+    optional(:maximum_inroute_duration, type: Integer, desc: 'Maximum in route duration of this particular shipment (Must be feasible !)')
+    optional(:sticky_vehicle_ids, type: Array[String], desc: 'Defined to which vehicle the shipment is assigned', coerce_with: ->(val) { val.is_a?(String) ? val.split(/,/) : val })
+    optional(:skills, type: Array[String], desc: 'Particular abilities required by a vehicle to perform this shipment', coerce_with: ->(val) { val.is_a?(String) ? val.split(/,/) : val })
+    requires(:pickup, type: Hash, allow_blank: false, desc: 'Activity of collection') do
+      use :vrp_request_activity
+    end
+    requires(:delivery, type: Hash, allow_blank: false, desc: 'Activity of drop off') do
+      use :vrp_request_activity
+    end
+    optional(:direct, type: Boolean, default: false, desc: 'When activated, vehicle should go directly to delivery point after pick up')
+    optional(:quantity_ids, type: String, documentation: { hidden: true }, desc: 'Quantities to consider, CSV front only')
+    optional(:quantities, type: Array, desc: 'Define the entities which are taken and dropped') do
+      use :vrp_request_quantity
+    end
+    mutually_exclusive :quantity_ids, :quantities
+  end
+end
+
+module VrpShared
+  extend Grape::API::Helpers
+
+  params :vrp_request_activity do
```

Comment:
Move to VrpMissions ?

Reply:
I centralize in VrpShared the objects which are used by multiple others + it helps to comply with rubocop rules on module sizes